modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/freelist/FreeList.java

```diff
@@ -21,30 +21,27 @@
 import org.apache.ignite.internal.lang.IgniteInternalCheckedException;
 import org.apache.ignite.internal.logger.IgniteLogger;
 import org.apache.ignite.internal.pagememory.Storable;
-import org.apache.ignite.internal.pagememory.metric.IoStatisticsHolder;
 import org.apache.ignite.internal.pagememory.util.PageHandler;
 
 /**
  * Free list.
  */
-public interface FreeList<T extends Storable> {
+public interface FreeList {
     /**
      * Inserts a row.
      *
      * @param row Row.
-     * @param statHolder Statistics holder to track IO operations.
      * @throws IgniteInternalCheckedException If failed.
      */
-    void insertDataRow(T row, IoStatisticsHolder statHolder) throws IgniteInternalCheckedException;
+    void insertDataRow(Storable row) throws IgniteInternalCheckedException;
```

Comment:
Why did you remove stat holder? Let's leave it for now. If you want to clean the code, then please do it in a separate PR, this one is already big enough.